gamemodes/terrortown/gamemode/server/sv_network_sync.lua

```diff
@@ -0,0 +1,677 @@
+--
+-- @param any|table path The path that this meta data is associated with
+-- @param table|nil metadata The metadata table
+function TTT2NET:SetMetaData(path, metadata)
+	-- Convert path with single key to table
+	if not istable(path) then
+		path = { path }
+	end
+
+	-- Retrieve the stored metadata
+	local storedMetaData = table.GetWithPath(data_store_metadata, path)
+
+	-- If the metadata is already stored, do nothing
+	if self:NetworkMetaDataTableEqual(storedMetaData, metadata) then return end
+
+	-- Insert data to metadata table
+	table.SetWithPath(data_store_metadata, path, metadata)
+	-- Set value of the data field to nil
+	table.SetWithPath(data_store, path, nil)
+
+	-- Sync new meta information to all clients that are already connected and received a full state!
+	self:SendMetaDataUpdate(path)
+end
+
+---
+-- This will set the metadata for a specific path and prepend the path with the "global" keyword.
+-- For more information look at {@TTT2NET:SetMetaData}.
+--
+-- @param any|table path The path this meta data is associated with (already includes the "global" keyword)
+-- @param table|nil metadata The metadata table
+function TTT2NET:SetMetaDataGlobal(path, metadata)
+	-- Convert path with single key to table
+	if not istable(path) then
+		path = { path }
+	end
+
+	-- Add the prefix for the correct table
+	table.insert(path, 1, "global")
+
+	self:SetMetaData(path, metadata)
+end
+
+---
+-- This will set the metadata for a specific path on a player and prepend needed keywords.
+-- For more information look at {@TTT2NET:SetMetaData}.
+--
+-- @param any|table path The path this meta data is associated with (already includes the needed keywords)
+-- @param table|nil metadata The metadata table
+function TTT2NET:SetMetaDataOnPlayer(path, metadata, ply)
+	-- Convert path with single key to table
+	if not istable(path) then
+		path = { path }
+	end
+
+	-- Add the prefix for the correct table with the specific player
+	table.insert(path, 1, "players")
+	table.insert(path, 2, ply:EntIndex())
+
+	self:SetMetaData(path, metadata)
+end
+
+---
+-- This is used to set a value in the data table for a specific path, it can take an additional meta data object,
+-- which represents the type of the data to be set. This will automatically create the metadata entry if it does not
+-- exist and is provided as a parameter. This can only be left empty, when the path already has a valid metadata entry.
+-- This will automatically synchronize the new data to the clients.
+-- When the additional client parameter is set, the data will be saved as an override for that specific client/entity.
+--
+-- @param any|table path The path to set the data for
+-- @param table|nil meta The metadata for the path
+-- @param any value The value to save
+-- @param Entity|nil client The client/entity to set this value for (overrides the default value)
+function TTT2NET:Set(path, meta, value, client)
+	local clientId = client and client:EntIndex() or nil
```

Comment:
client theoretically is a thing for the client realm (especially LocalPlayer). Maybe player would match better?

Reply:
And maybe add IsValid for client because it can be "Not nil" but invalid

Reply:
I used client, to emphasize on the difference between setting data on "players" and overriding data on "clients" as "clients" is a term closer to the network related code